utils.groovy

```diff
@@ -233,4 +247,97 @@ def getDockerTag() {
   return "${IS_PRO ? 'pro-' : ''}${getDockerBuildOs(env.OS)}-${env.ARCH}-${RSTUDIO_VERSION_FLOWER}"
 }
 
+/**
+  * Rename the file on the filesystem
+  */
+def renameFile(String dir, String match) {
+  def file = sh(script: "basename `ls ${dir}/${match}`", returnStdout: true).trim()
+  def renamedFile = file.replace('-relwithdebinfo', '')
+  sh "mv ${dir}/${file} ${dir}/${renamedFile}"
+  return renamedFile
+}
+
+/**
+  * Rename the file on the filesystem
+  */
+def renameTarFile(String dir) {
+  if ((FLAVOR == "Desktop") || (FLAVOR == "Electron")) {
+    return renameFile(dir, "*.tar.gz")
+  }
+
+  return ""
+}
+
+def getAgentLabel(String arch) {
+  if (arch == 'arm64') {
+    return 'linux && arm64 && 4x'
+  }
+
+  return 'linux-4x && x86_64'
+}
+
+// Use a function to keep the pipeline length down.
+def shouldBuild(boolean isDaily, boolean isPro) {
+  def matchesFilter = ((params.OS_FILTER == env.OS ||  params.OS_FILTER == 'all') &&
+    (params.ARCH_FILTER == env.ARCH ||  params.ARCH_FILTER == 'all') &&
+    (params.FLAVOR_FILTER == env.FLAVOR ||  params.FLAVOR_FILTER == 'all'))
+
+  // Filter hourlies based on https://github.com/rstudio/rstudio-pro/issues/4143#issuecomment-1362142399
+  def inHourlySubset = true
+  if (!isDaily) {
+    if (isPro) {
+      // build an x86_64 rhel8 WB, x86_64 rhel9 QT RDP, and an arm64 jammy Electron RDP 
+      inHourlySubset = ((env.OS == 'rhel8' && env.ARCH == 'x86_64' && env.FLAVOR == 'Server') ||
+        (env.OS == 'centos7' && env.ARCH == 'x86_64' && env.FLAVOR == 'Desktop') || 
+        (env.OS == 'jammy' && env.ARCH == 'arm64' && env.FLAVOR == 'Electron'))
+    } else {
+      // build an arm64 rhel9 Desktop and an x86_64 focal Server
+      inHourlySubset = ((env.OS == 'rhel9' && env.ARCH == 'arm64' && env.FLAVOR == 'Electron') ||
+        (env.OS == 'focal' && env.ARCH == 'x86_64' && env.FLAVOR == 'Server'))
+    }
+  }
```

Comment:
I realize I'm nit-picking code we already had, and I hate to be that guy but do you think the legibility of this could benefit from a few early returns?